### PR TITLE
Add TrustyAIService resource

### PR DESCRIPTION
**Pull request by adolfo-ab**
##### Short description:
Adds the TrustyAIService resource

##### More details:
This resource allows users to deploy TrustyAIService, which provides explainability features for AI models

### CODE REVIEW

File: ocp_resources/trustyai_service.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    """
+
+    api_group: str = NamespacedResource.ApiGroup.TRUSTYAI_OPENDATAHUB_IO
+
+    def __init__(
+        self,
+        storage: Dict[str, Any],
+        data: Dict[str, Any],
+        metrics: Dict[str, Any],
+        replicas: Optional[int] = 1,
```
Comment (myakove) [inline ocp_resources/trustyai_service.py:17]: Please do not set any defaults, if the user sent we set it, else we let k8s to handle it and set the default.

File: ocp_resources/trustyai_service.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+class TrustyAIService(NamespacedResource):
+    """
+    https://github.com/trustyai-explainability/trustyai-service-operator/blob/main/api/v1alpha1/trustyaiservice_types.go
+    """
+
+    api_group: str = NamespacedResource.ApiGroup.TRUSTYAI_OPENDATAHUB_IO
+
+    def __init__(
+        self,
+        storage: Dict[str, Any],
```
Comment (myakove) [inline ocp_resources/trustyai_service.py:14]: all should be Optional, the user can pass yaml file instead.

File: ocp_resources/trustyai_service.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        self.data = data
+        self.metrics = metrics
+        self.replicas = replicas
+        self.image = image
+        self.tag = tag
+
+    def to_dict(self) -> None:
+        super().to_dict()
+
+        if not self.yaml_file:
```
Comment (myakove) [inline ocp_resources/trustyai_service.py:44]: here you can raise if some mandatory args are not passed.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/trustyai_service.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        **kwargs,
+    ) -> None:
+        """
+        TrustyAIService object
+
+        Args:
+            storage (Optional[Dic[str, Any]]): TrustyAIService storage config (format, folder and size).
+            data (Optional[Dict[str, Any]]): TrustyAIService data config (filename and format).
+            metrics (Optional[Dict[str, Any]]): TrustyAIService metrics config.
+            replicas (Optional[int], default: 1): Number of replicas for the TrustyAIService.
```
Comment (rnetser) [inline ocp_resources/trustyai_service.py:29]: please remove the default value as it is not set in the wrappers' code but by the api

File: ocp_resources/trustyai_service.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        self.replicas = replicas
+        self.image = image
+        self.tag = tag
+
+    def to_dict(self) -> None:
+        super().to_dict()
+
+        if not self.yaml_file:
+            if not (self.storage or self.data or self.metrics):
+                raise MissingRequiredArgumentError(argument="'storage' or 'data' or 'metrics'")
```
Comment (rnetser) [inline ocp_resources/trustyai_service.py:46]: please remove the `optional` from the docstring for these attributes

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/trustyai_service.py (hunk 5 above)
Comment (myakove) [reply ocp_resources/trustyai_service.py:46]: They are still optional from the function perspective and should stay as optional in the docsting.

Comment (rnetser) [reply ocp_resources/trustyai_service.py:46]: so `mandatory if not passing yaml_file` should be added to the relevant places in the docstring

Comment (myakove) [reply ocp_resources/trustyai_service.py:46]: ok

**Review by myakove — APPROVED** [review]

**Pull request merged**